static/app/views/dashboardsV2/dashboard.tsx

```diff
@@ -199,11 +202,13 @@ class Dashboard extends Component<Props, State> {
   }
 
   async addNewWidget() {
-    const {api, organization, newWidget, handleAddCustomWidget} = this.props;
+    const {api, organization, newWidget, handleAddCustomWidget, onSetNewWidget} =
+      this.props;
     if (newWidget) {
       try {
         await validateWidget(api, organization.slug, newWidget);
         handleAddCustomWidget(newWidget);
+        onSetNewWidget && onSetNewWidget();
```

Comment:
I think this is equivalent?
```suggestion
        onSetNewWidget?.();
```